Description: Fix failure to import _lxc on armhf
 Recent testing on Ubuntu armhf showed that the python module was
 failing to import. After some time tracking the issue down, the problem
 was identified as being a non-terminated list of get/setters.
 .
 This commit fixes that issue as well as a few other potential ones that
 were identified during debugging.

Author: Stéphane Graber <stgraber@ubuntu.com>
Origin: upstream
Bug-Ubuntu: https://launchpad.net/bugs/1159817
Forwarded: not-needed
Reviewed-By: Stéphane Graber <stgraber@ubuntu.com>
Last-Update: 2013-03-26

--- lxc-0.9.0~rc1.orig/src/python-lxc/lxc.c
+++ lxc-0.9.0~rc1/src/python-lxc/lxc.c
@@ -96,13 +96,13 @@ Container_init(Container *self, PyObject
 }
 
 static PyObject *
-get_default_config_path(Container *self, PyObject *args, PyObject *kwds)
+LXC_get_default_config_path(PyObject *self, PyObject *args)
 {
     return PyUnicode_FromString(lxc_get_default_config_path());
 }
 
 static PyObject *
-get_version(Container *self, PyObject *args, PyObject *kwds)
+LXC_get_version(PyObject *self, PyObject *args)
 {
     return PyUnicode_FromString(lxc_get_version());
 }
@@ -496,6 +496,7 @@ static PyGetSetDef Container_getseters[]
      (getter)Container_state, 0,
      "Container state",
      NULL},
+    {NULL, NULL, NULL, NULL, NULL}
 };
 
 static PyMethodDef Container_methods[] = {
@@ -595,7 +596,7 @@ static PyMethodDef Container_methods[] =
      "\n"
      "Wait for the container to reach a given state or timeout."
     },
-    {NULL}  /* Sentinel */
+    {NULL, NULL, 0, NULL}
 };
 
 static PyTypeObject _lxc_ContainerType = {
@@ -641,9 +642,9 @@ PyVarObject_HEAD_INIT(NULL, 0)
 };
 
 static PyMethodDef LXC_methods[] = {
-    {"get_default_config_path",  (PyCFunction)get_default_config_path, METH_NOARGS,
+    {"get_default_config_path", (PyCFunction)LXC_get_default_config_path, METH_NOARGS,
      "Returns the current LXC config path"},
-    {"get_version",  (PyCFunction)get_version, METH_NOARGS,
+    {"get_version", (PyCFunction)LXC_get_version, METH_NOARGS,
      "Returns the current LXC library version"},
     {NULL, NULL, 0, NULL}
 };
